Base/splitter.h

```diff
@@ -0,0 +1,42 @@
+#ifndef SPLITTER_H__
+#define SPLITTER_H__
+
+struct SplitterRecord
+{
+  double threshold;       // Threshold to split at.
+  double improvement;     // Impurity improvement given parent node.
+  double impurity_left;   // Impurity of the left split.
+  double impurity_right;  // Impurity of the right split.
+};
+
+class SplitterInterface{
```

Comment:
I think we can drop the  `Interface` suffix. It is pretty clear from the virtual methods that this is an interface class.